presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFiltersCollector.java

```diff
@@ -79,33 +84,91 @@ public void collectDynamicFilterDomains(Map<DynamicFilterId, Domain> dynamicFilt
     // Called during TableScan planning (no need to be synchronized as local planning is single threaded)
     public DynamicFilter createDynamicFilter(List<Descriptor> descriptors, Map<Symbol, ColumnHandle> columnsMap)
     {
-        Multimap<DynamicFilterId, Symbol> symbolsMap = descriptors.stream()
-                .collect(toImmutableSetMultimap(Descriptor::getId, descriptor -> Symbol.from(descriptor.getInput())));
+        Multimap<DynamicFilterId, Descriptor> descriptorMap = descriptors.stream()
+                .collect(toImmutableSetMultimap(Descriptor::getId, Function.identity()));
```

Comment:
Could this be `toImmutableListMultimap`  instead ? Are duplicate `Descriptor` possible here ?

Reply:
IIUC, it may only happen if the same DF function appears twice in the filter expression:
https://github.com/prestosql/presto/blob/8c6901c0c94593c9e5f1d65ea471697248f44dc0/presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java#L1370
I think that this should not happen - so I changed the code to use `toImmutableListMultimap`.